core-bundle/src/Cron/Cron.php

```diff
@@ -175,4 +203,17 @@ function ($reason) use ($cron): void {
 
         Utils::settle($promises)->wait();
     }
+
+    /**
+     * Checks if a database connection can be established and the table exist.
+     */
+    private function canRunDbQuery(Connection $connection): bool
+    {
+        try {
+            return $connection->isConnected()
```

Comment:
@fritzmg why exactly do we check for `isConnected()`? This would mean that if there wasn't any previous query, crons would never be run? 🤔

Reply:
It's always needed when `$this->cron->run()` is executed. No matter where it's executed from. The responsibility should be in the cron service.

Reply:
But the purpose was to prevent an exception from happening in regular requests and thus it is only relevant in the `kernel.terminate` lisener. If you run the web or CLI cron an exception is fine.

Reply:
I've pushed the previous state again, as you can see, tests are somehow failing. The cron scheduler seems to be executed still.

Reply:
Does it work locally? i.e., if you remove the database credentials (or make them purposefully wrong) and you access the front end - will you see a separate exception from the `kernel.terminate` event regarding the database?

Reply:
I found the problem: The issue is that our functional tests do not in fact run on a 100% complete setup. They only insert `tl_page` entries etc. So the cron jobs are now triggered and thus fail at e.g. the `PurgePreviewLinksCron`. We previously disabled running crons during the tests which is now not possible anymore.
Seems like we still need the setting for our own tests? Any other ideas?

Reply:
May be we should still introduce a bundle config, with which you can forcibly disable the `CommandSchedulerListener`? Then we could disable it for our tests and in the real world you can also disable it, if you want to (but you don't have to).

Reply:
Alternative proposal: We move the setting to `contao.cron.web_listener` that allows to take on the current values `true` and `false`. We can force disabling it in our tests as we do now then and someone can also force enable it, if they want to have the `web` scope for some reason. New would be an `auto` configuration which implements the logic of this PR and becomes the new default value.
That way, we can support it all.

Reply:
Sounds good to me :)

Reply:
+1 for a true/false/auto setting ❤️